src/firefly/js/Firefly.js

```diff
@@ -65,35 +86,81 @@ export const Templates = {
  *
  */
 
+/** @type {AppProps} */
+const defAppProps = {
+    div: 'app',
+    template: undefined,        // don't set a default value for this.  it's also used as a switch for API vs UI mode
+    appTitle: '',
+    showUserInfo: false,
+    showViewsSwitch: false,
+    rightButtons: undefined,
+
+    menu: [
+        {label:'Images', action:'ImageSelectDropDownCmd'},
+        {label:'Catalogs', action:'IrsaCatalogDropDown'},
+        {label:'Charts', action:'ChartSelectDropDownCmd'},
+        {label:'Upload', action: 'FileUploadDropDownCmd'},
+        //{label:'Workspace', action: 'WorkspaceDropDownCmd'}
+    ],
+};
 
-const defConfigOptions = {
+/** @type {FireflyOptions} */
+const defFireflyOptions = {
     MenuItemKeys: {},
     imageTabs: undefined,
     irsaCatalogFilter: undefined,
     catalogSpacialOp: undefined,
     imageMasterSources: ['ALL'],
-    imageMasterSourcesOrder: '',
+    imageMasterSourcesOrder: undefined,
     workspace : { showOptions: false},
-    hips : {useForCoverage: true, useForImageSearch: true,
-            hipsSources: 'all',
-            defHipsSources: {source: 'irsa', label: 'Featured'},
-            mergedListPriority: 'irsa'}
+
+    charts: {
+        // TODO: need to define all options with defaults here.
```

Comment:
maxRowsForScatter: 5000, // maximum table rows, for which scatter plot can be created
multitrace: true,  // deprecated - by default we use multi-trace architecture
singleTraceUI: false, // by default we support multi-trace in UI
ui: {HistogramOptions: {fixedAlgorithm: undefined}} // by default we allow both "uniform binning" and "bayesian blocks"